src/cpp/rtps/DataSharing/DataSharingNotification.cpp

```diff
@@ -50,101 +50,36 @@ std::shared_ptr<DataSharingNotification> DataSharingNotification::open_notificat
     return notification;
 }
 
-void DataSharingNotification::destroy()
-{
-    if (owned_)
-    {
-        // We cannot destroy the objects in the SHM, as the Writer may still be using them.
-        // We just remove the segment, and when the Writer closes it, it will be removed from the system.
-        segment_->remove(segment_name_);
-
-        owned_ = false;
-    }
-    else
-    {
-        logError(HISTORY_DATASHARING_LISTENER, "Trying to destroy non-owned notification segment " << segment_name_);
-    }
-}
-
 bool DataSharingNotification::create_and_init_notification(
         const GUID_t& reader_guid,
         const std::string& shared_dir)
 {
-    segment_id_ = reader_guid;
-    segment_name_ = generate_segment_name(shared_dir, reader_guid);
-
-    uint32_t per_allocation_extra_size = Segment::compute_per_allocation_extra_size(
-        alignof(Notification), DataSharingNotification::domain_name());
-    uint32_t segment_size = static_cast<uint32_t>(sizeof(Notification)) + per_allocation_extra_size;
-
-    //Open the segment
-    Segment::remove(segment_name_);
-    try
-    {
-        segment_ = std::unique_ptr<Segment>(
-            new Segment(boost::interprocess::create_only,
-            segment_name_,
-            segment_size + fastdds::rtps::SharedMemSegment::EXTRA_SEGMENT_SIZE));
-    }
-    catch (const std::exception& e)
-    {
-        logError(HISTORY_DATASHARING_LISTENER, "Failed to create segment " << segment_name_
-                                                                           << ": " << e.what());
-        return false;
-    }
-
-    try
-    {
-        // Alloc and initialize the Node
-        notification_ = segment_->get().construct<Notification>("notification_node")();
-        notification_->new_data.store(false);
-    }
-    catch (std::exception& e)
-    {
-        Segment::remove(segment_name_);
-
-        logError(HISTORY_DATASHARING_LISTENER, "Failed to create listener queue " << segment_name_
-                                                                                  << ": " << e.what());
-        return false;
-    }
-
-    owned_ = true;
-    return true;
+    return create_and_init_shared_segment_notification<fastdds::rtps::SharedMemSegment>(reader_guid,
```

Comment:
Could this sometimes be `SharedFileSegment` ?

Reply:
Absolutely. This was lost on 88cde53. Corrected on bdc5d2e